frontend/src/queries/nodes/WebOverview/EvenlyDistributedRows.tsx

```diff
@@ -0,0 +1,66 @@
+import React, { useCallback, useEffect, useRef, useState } from 'react'
+import clsx from 'clsx'
+
+export const EvenlyDistributedRows = ({
+    children,
+    minWidthRems,
+    className,
+}: {
+    children: React.ReactNode[]
+    minWidthRems: number
+    className: string
+}): JSX.Element => {
+    const [rowLayout, setRowLayout] = useState<{ itemsPerRow: number; numRows: number }>()
+    const elementRef = useRef<HTMLDivElement>(null)
+
+    const updateSize = useCallback((): void => {
+        if (!elementRef.current) {
+            return
+        }
+        const pxPerRem = parseFloat(getComputedStyle(document.documentElement).fontSize)
+        const minWidthPx = minWidthRems * pxPerRem
+        const containerWidthPx = elementRef.current.offsetWidth
+
+        const maxItemsPerRow = Math.floor(containerWidthPx / minWidthPx)
+        // Distribute items evenly
+        // e.g. if we can have 4 elements per row and have 9 items
+        // prefer 3,3,3 to 4,4,1
+        const numRows = Math.ceil(children.length / maxItemsPerRow)
+        const itemsPerRow = Math.ceil(children.length / numRows)
```

Comment:
nit: What would happen with 13 items? If I understand correctly this code is to avoid orphans and it seems to break in this case. Doesn't need to be fixed if it works for your case, just tried to understand why we don't use flexbox here.

Reply:
Yes, 13 would be split in 4,4,4,1 either way. On a grid there isn't really a better option, as 4, 3, 3, 3 still has an orphaned item, just vertically (the right-most column only has one element). I could use flexbox to stretch the items so it'd look like
```
000111222333
444455556666
777788889999
AAAABBBBCCCC
```
but that's a different aesthetic than what I am going for :)

With 5 elements this component is useful though, as with flexbox you get this:
<img width="579" alt="Screenshot 2023-10-24 at 20 41 57" src="https://github.com/PostHog/posthog/assets/2056078/b7216fc4-29e9-4a3b-b92e-f33c800414f4">

whereas with grid I get this:
<img width="553" alt="Screenshot 2023-10-24 at 20 44 33" src="https://github.com/PostHog/posthog/assets/2056078/26fae207-ef6a-4e3c-93ab-2ba76ff9a71d">

I could use multiple flexboxes (one per row) to achieve a similar effect, but I chose to use grid as this is what it is good at.